parsers/deathloopvaccine.py

```diff
@@ -0,0 +1,286 @@
+#                   player deaths trigger the simulated process kill, or after any simulated
+#                   player death events "scroll off" the death loop monitoring window.
+#
+class DeathLoopVaccine(Parser):
+
+    """Tracks for DL symptoms"""
+
+    def __init__(self):
+
+        super().__init__()
+        self.name = 'deathloopvaccine'
+
+        # parameters that define a deathloop condition, i.e. D deaths in T seconds,
+        # with no player activity in the interim
+        # todo - make these configarable via the UI?
+        self.deathloop_deaths = config.data['deathloopvaccine']['deaths']
+        self.deathloop_seconds = config.data['deathloopvaccine']['seconds']
+
+        # list of death messages
+        # this will function as a scrolling queue, with the oldest message at position 0,
+        # newest appended to the other end.  Older messages scroll off the list when more
+        # than deathloop_seconds have elapsed.  The list is also flushed any time
+        # player activity is detected (i.e. player is not AFK).
+        #
+        # if/when the length of this list meets or exceeds deathloop_deaths, then
+        # the deathloop response is triggered
+        self._death_list = list()
+
+        # flag indicating whether the "process killer" gun is armed
+        self._kill_armed = True
+
+    def reset(self) -> None:
+        """
+        Utility function to clear the death_list and reset the armed flag
+
+        Returns:
+            None:
+        """
+        self._death_list.clear()
+        self._kill_armed = True
+
+    # main parsing logic here
+    def parse(self, timestamp: datetime, text: str) -> None:
+        """
+        Parse a single line from the logfile
+
+        Args:
+            timestamp: A datetime.datetime object, created from the timestamp text of the raw logfile line
+            text: The text following the everquest timestamp
+
+        Returns:
+            None:
+        """
+
+        self.check_for_death(timestamp, text)
+        self.check_not_afk(timestamp, text)
+        self.deathloop_response()
+
+    def check_for_death(self, timestamp: datetime, text: str) -> None:
+        """
+        check for indications the player just died, and if we find it,
+        save the message for later processing
+
+        Args:
+            timestamp: A datetime.datetime object, created from the timestamp text of the raw logfile line
+            text: The text following the everquest timestamp
+
+        Returns:
+            None:
+        """
+
+        # reconstruct the full logfile line
+        # this is a bit counter-intuitive, but the rest of the logic in this function was
+        # developed assuming the line was the full line, including the time stamp
+        line = f'[{timestamp.strftime("%a %b %d %H:%M:%S %Y")}] ' + text
+
+        # cut off the leading date-time stamp info
+        trunc_line = line[27:]
```

Comment:
Is this different from `text`?
IDK it feels like you likely could just rework the rest of the function to use the timestamp rather than having to reconstruct this other line, too

Reply:
Yeah.  Two reasons:
1. I did spend a fair amount of time getting them working for the standalone DLV program, which was written assuming the full, unchanged EQ log line was passed as input, and then the functions disect it into a timestamp and the rest of the line.  It's time consuming and error prone to rewrite the whole rest of the function for no real utility gain.
2. The bigger reason = the utility keeps a queue of time-stamped death messages, and then uses the time stamp portion of each death line to scroll them off when they get too old.  Having a separate timestamp and rest of the line means I would have to rewrite it to completely change the logic somehow, either with having two containers, or only keeping the timestamps but not the death message, or something.  Again, a bunch of error prone work for no real utility gain.

So I just took the passed data, put it back together, and then copied my already-developed functions in unchanged.

Reply:
I think I'd like to take a crack at doing it a slightly different way that doesn't require this kind of text mangling. Date-time item expiration is a pretty common pattern and I do it already in some other places.

Reply:
Ok.  I did rework it some to make use of the passed (timestamp, text) data where possible, so it's now at least a bit more sensible than mindless combination immediately followed by separation, which is what it was doing before.  The current logic does however keep the bit about recombining them for retention, expiration, and reporting purposes, so if you want to rework that bit for consistency you can.